### PR TITLE
rec: enable qname minimization

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Enablq qname minimization by default.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [X] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
Not yet ready for primetime. e.g. reggresion test failure that need to be investigated.

**Comment by omoerbeek** [comment]
All tests succeed now.

There is one rather fundamental change in this PR,  and that is the handling of forwarding in `syncres.cc`. To be discussed....

File: regression-tests.recursor-dnssec/test_Protobuf.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def checkProtobufBase(self, msg, protocol, query, initiator, normalQueryResponse
         self.assertEquals(msg.id, query.id)
         self.assertTrue(msg.HasField('inBytes'))
         if normalQueryResponse:
-            # compare inBytes with length of query/response
+            # compare inBytes with length off query/response
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_Protobuf.py]: Wasn't `of` actually the right version?

File: regression-tests.recursor-dnssec/test_Protobuf.py [hunk 2]
```diff
@@ -351,7 +351,10 @@ class OutgoingProtobufDefaultTest(TestRecursorProtobuf):
 
     _confdir = 'OutgoingProtobufDefault'
     _config_template = """
-auth-zones=example=configs/%s/example.zone""" % _confdir
+    # Switch of QName Minimization, it generates much more protobuf messages
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_Protobuf.py]:
```suggestion
    # Switch off QName Minimization, it generates much more protobuf messages
```

File: regression-tests.recursor-dnssec/test_Protobuf.py [hunk 3]
```diff
@@ -380,7 +383,10 @@ class OutgoingProtobufNoQueriesTest(TestRecursorProtobuf):
 
     _confdir = 'OutgoingProtobufNoQueries'
     _config_template = """
-auth-zones=example=configs/%s/example.zone""" % _confdir
+    # Switch of QName Minimization, it generates much more protobuf messages
```
Comment (rgacogne) [inline regression-tests.recursor-dnssec/test_Protobuf.py]:
```suggestion
    # Switch off QName Minimization, it generates much more protobuf messages
```

File: regression-tests.recursor-dnssec/test_Protobuf.py (hunk 1 above)
Comment (Habbie) [reply regression-tests.recursor-dnssec/test_Protobuf.py]: Indeed

Comment (omoerbeek) [reply regression-tests.recursor-dnssec/test_Protobuf.py]: Yeah, only the other was correct, thanks

**Pull request merged**